src/app/posts/index/index.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Router } from '@angular/router';
+import { Post } from '../../post';
```

Comment:
Separate the imports by groups. For ex: imports from angular packages, then 1 empty line and then import your packages:

Import HttpClient
Import Router

Import Post